pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt

```diff
@@ -74,33 +70,39 @@ val AWSSQSPlugin = createApplicationPlugin(
         SchemaValidation.logger.info("Consuming messages from AWS SQS")
         CoroutineScope(Dispatchers.IO).launch {
             while (true) {
+                var response: ReceiveMessageResponse? = null
                 try {
                     val receiveMessageRequest = ReceiveMessageRequest {
                         this.queueUrl = queueUrl
                         maxNumberOfMessages = 5
                     }
-                    val response = sqsClient.receiveMessage(receiveMessageRequest)
+                    response = sqsClient.receiveMessage(receiveMessageRequest)
                     response.messages?.forEach { message ->
                         SchemaValidation.logger.info("Received message from AWS SQS: ${message.body}")
-
                         message.body?.let { AWSSQSProcessor().validateMessage(it) }
-                        val deleteMessageRequest = DeleteMessageRequest {
-                            this.queueUrl = queueUrl
-                            this.receiptHandle = message.receiptHandle
-                        }
-                        sqsClient.deleteMessage(deleteMessageRequest)
-                        SchemaValidation.logger.info("Successfully Deleted processed message from AWS SQS")
-
                     }
                 } catch (e: AwsServiceException) {
                     SchemaValidation.logger.error("Something went wrong while processing the request ${e.message}")
+                    throw e
                 } catch (e: Exception) {
                     SchemaValidation.logger.error("AWS service exception occurred: ${e.message}")
+                    throw e
+                }finally {
+                    response?.messages?.forEach { message ->
+                        try {
+                            val deleteMessageRequest = DeleteMessageRequest {
+                                this.queueUrl = queueUrl
+                                this.receiptHandle = message.receiptHandle
+                            }
+                            sqsClient.deleteMessage(deleteMessageRequest)
+                            SchemaValidation.logger.info("Successfully Deleted processed message from AWS SQS")
+                        }catch (e: Exception) {
+                            SchemaValidation.logger.error("Something went wrong while deleting the report from the queue ${e.message}")
+                        }
```

Comment:
Do we need a throw here? And a retry maybe? Otherwise it will be a false positive message retained in the queue isn't it?

Reply:
Good point. I will add throw here and look into retry

Reply:
This will be addressed in another story for now it will go as-is